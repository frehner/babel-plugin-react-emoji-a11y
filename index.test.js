const babel = require("babel-core");
const plugin = require("./index");

fit(`should be able to find an emoji`, () => {
  const example = `
    function Blah() {
      return (
        <div role="button">
          test testing test 🐽 blah bal
        </div>
      )
    }

    class Something extends React.component {
      render() {
        return (
          <span aria-label="pig nose">🐽</span>
        )
      }
    }
  `;
  const { code } = babel.transform(example, { plugins: [plugin] });
  console.log(code);
  // expect(code).toMatchSnapshot()
});

it(`should not warn if they did it correctly`, () => {
  const example = `
    function Correct() {
      return (
        <span aria-label="pig nose" role="img">🐽</span>
      )
    }
  `;

  const { code } = babel.transform(example, { plugins: [plugin] });
});
