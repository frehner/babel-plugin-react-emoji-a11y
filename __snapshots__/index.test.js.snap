// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add "aria-label" to a span that doesn't have it 1`] = `
"
class AlmostThere extends React.component {
  render() {
    return <span role=\\"img\\" aria-label=\\"pig nose\\">🐽</span>;
  }
}"
`;

exports[`should add "role" to a span that doesn't have it 1`] = `
"
class AlmostThere extends React.component {
  render() {
    return <span aria-label=\\"pig nose\\" role=\\"img\\">🐽</span>;
  }
}"
`;

exports[`should do nothing if the emoji is accessible already 1`] = `
"
function Correct() {
  return <span aria-label=\\"shouldn't change\\" role=\\"img\\">🐽</span>;
}"
`;

exports[`should fix emojis that are in accessible spans but with other text 1`] = `
"
function NotQuiteButClose() {
  return <span aria-label=\\"shouldn't change\\" role=\\"img\\">asd <span aria-label=\\"pig nose\\" role=\\"img\\">🐽</span> asdf</span>;
}"
`;

exports[`should wrap an emoji that doesn't have any accessibility span 1`] = `
"
function Bad() {
  return <div>
          test <span aria-label=\\"pig nose\\" role=\\"img\\">🐽</span> blah
        </div>;
}"
`;
